pkg/util/limitrange/limitrange.go

```diff
@@ -105,16 +110,27 @@ func calculateMainContainersResources(containers []corev1.Container) corev1.Reso
 }
 
 func calculateInitContainersResources(initContainers []corev1.Container) corev1.ResourceList {
+	incrementalSum := calculateSidecarContainersIncrementalSum(initContainers)
 	total := corev1.ResourceList{}
 	for i := range initContainers {
-		sumSidecarContainers := calculateSidecarContainersResources(initContainers[:i])
-		initContainerUse := resource.MergeResourceListKeepSum(sumSidecarContainers, initContainers[i].Resources.Requests)
-
+		initContainerUse := resource.MergeResourceListKeepSum(incrementalSum[i], initContainers[i].Resources.Requests)
 		total = resource.MergeResourceListKeepMax(total, initContainerUse)
 	}
 	return total
 }
 
+func calculateSidecarContainersIncrementalSum(initContainers []corev1.Container) []corev1.ResourceList {
+	results := make([]corev1.ResourceList, len(initContainers))
+	lastResult := corev1.ResourceList{}
+	for i := 1; i < len(initContainers); i++ {
+		if isSidecarContainer(initContainers[i-1]) {
+			lastResult = resource.MergeResourceListKeepSum(results[i-1], initContainers[i-1].Resources.Requests)
+		}
+		results[i] = lastResult
+	}
+	return results
+}
+
```

Comment:
I think this can be simplified a bit, also with O(1) memory complexity 
```
def calculateInitContainerResources():
  sideCarRunningUsage = resourceUsage()
  maxInitContainerUsage = resourceUsage()
  for container in initContainers:
    if isSideCar(container):
      sideCarRunningUsage = mergeKeepSum(sideCarRunningUsage, usage(container))
    else:
      initContainerUse = usage(container) + sideCarRunningUsage
      maxInitContainerUsage = mergeKeepMax(maxInitContainerUsage, initContainerUse)
  return maxInitContainerUsage

```

Reply:
Done. Thanks :)